src/xo-app/settings/plugins/index.js

```diff
@@ -120,7 +121,13 @@ class Plugin extends Component {
     this.refs.pluginInput.value = this.props.configurationPresets[configName]
   }
 
-  _test = async () => testPlugin(this.props.id, this.refs.testInput.value)
+  _test = async () => {
+    try {
+      await testPlugin(this.props.id, this.refs.testInput.value)
+    } catch (err) {
+      alert('You have an error!', err.stack)
```

Comment:
The stack is irrelevant because it contains information about where the error was parsed by the client not where it was generated in the server.

What you need to display is:

- its code
- its message
- its data (an optional JavaScript value which can be an object, therefore you should show its JSON representation)